src/ra_server_proc.erl

```diff
@@ -1717,9 +1726,11 @@ handle_tick_metrics(State) ->
     _ = ets:insert(ra_metrics, Metrics),
     State.
 
-can_execute_locally(RaftState, TargetNode, State) ->
+can_execute_locally(RaftState, TargetNode,
+                    #state{server_state = ServerState} = State) ->
+    {Voter, _} = ra_server:voter_status(ServerState),
     case RaftState of
-        follower ->
+        follower when Voter == voter ->
```

Comment:
The next clause (`leader when TargetNode =/= node()`) will probably need to take voting status into account else the leader may not execute the effect instead of the non voter.

Reply:
coordination_SUITE has some local tests so that may be a good place to put a test for this.

Reply:
Done via a new local query, tests included 👍